File: backend/apps/hazard/serializers/hazard.py
from rest_framework import serializers
from apps.hazard.models import HazardCurve, HazardModel


class HazardModelSerializer(serializers.ModelSerializer):
    class Meta:
        model = HazardModel
        fields = ["id", "name", "organization", "version", "description", "config_hash", "is_active"]


class HazardCurveSerializer(serializers.ModelSerializer):
    latitude = serializers.FloatField()
    longitude = serializers.FloatField()
    hazard_model = HazardModelSerializer(read_only=True)

    class Meta:
        model = HazardCurve
        fields = [
            "id",
            "intensity_measure_type",
            "latitude",
            "longitude",
            "vs30_m_s",
            "pga_values",
            "annual_exceedance_rates",
            "hazard_model",
        ]

File: frontend/types/hazard.ts
export interface HazardModel {
  id: string;
  name: string;
  organization: string;
  version: string;
  description?: string;
  config_hash?: string;
  is_active: boolean;
}

export interface HazardCurve {
  id: string;
  intensity_measure_type: string;
  latitude: number;
  longitude: number;
  vs30_m_s: number;
  pga_values: number[];
  annual_exceedance_rates: number[];
  hazard_model: HazardModel;
}
